src/App.jsx

```diff
@@ -14,33 +16,147 @@
-        Reverse
-      </button>
+    if (reversePressed) {
+      switch (type) {
+        case SORT_ALPHABETICALLY:
+          newGoods.sort((good1, good2) => good2.localeCompare(good1));
+          setlengthPressed(false);
+          break;
+        case SORT_BY_LENGTH:
+          newGoods = [...sortedGoods].sort(
+            (good1, good2) => good2.length - good1.length,
+          );
+          setalphPressed(false);
+
+          break;
+        case REVERSE:
+          newGoods = [...sortedGoods].reverse();
+          break;
+        case RESET:
+          newGoods = [...goodsFromServer];
+          setreversePressed(false);
+          setalphPressed(false);
+          setlengthPressed(false);
+          break;
+        default:
+          break;
+      }
+    } else {
+      switch (type) {
+        case SORT_ALPHABETICALLY:
+          newGoods.sort((good1, good2) => good1.localeCompare(good2));
+          setlengthPressed(false);
+          break;
+        case SORT_BY_LENGTH:
+          newGoods.sort((good1, good2) => good1.length - good2.length);
+          setalphPressed(false);
+
+          break;
+        case REVERSE:
+          newGoods = [...sortedGoods].reverse();
+          break;
+        case RESET:
+          newGoods = [...goodsFromServer];
+          setreversePressed(false);
+          setalphPressed(false);
+          setlengthPressed(false);
+          break;
+        default:
+          break;
+      }
+    }
+
+    setsortedGoods(newGoods);
+    setSort(type);
+  };
 
-      <button type="button" className="button is-danger is-light">
+  const resetButton = () => {
+    return (
+      <button
+        type="button"
+        className="button is-danger"
+        onClick={() => {
+          handleSort(RESET);
+        }}
+      >
         Reset
       </button>
-    </div>
+    );
+  };
+
+ 
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': alphPressed !== true,
+            NOT_ACTIVE_CLASS: sort !== SORT_ALPHABETICALLY,
+          })}
+          onClick={() => {
+            handleSort(SORT_ALPHABETICALLY);
+            setalphPressed(!alphPressed);
+          }}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          onClick={() => {
+            handleSort(SORT_BY_LENGTH);
+            setlengthPressed(!lengthPressed);
+          }}
+          className={classNames('button is-success', {
+            'is-light': lengthPressed !== true,
+            NOT_ACTIVE_CLASS: sort !== SORT_BY_LENGTH,
```

Comment:
The `NOT_ACTIVE_CLASS` is used here but it is not defined in the provided code. Ensure that this class is defined or imported correctly.